src/main/java/com/andreamazzarella/contact_manager/TelephoneNumber.java

```diff
@@ -7,27 +7,32 @@
     private final String number;
 
     public TelephoneNumber(String number) {
-        number = removeAllWhiteSpace(number);
+        checkValidityOfNumber(number);
 
-        if (isValidNumber(number)) {
-            this.number = number;
-        } else {
-            this.number = "0";
-        }
+        this.number = removeAllWhiteSpace(number);
     }
 
-    public String getValue() {
+    @Override
+    public String toString() {
         return number;
     }
 
-    private boolean isValidNumber(String number) {
+    private void checkValidityOfNumber(String number) {
+        if (!isValidNumber(number)) throw new IllegalArgumentException("Sorry, but the telephone number you entered is not valid.");
```

Comment:
Not sure about this validation belonging here. Few reasons;

1. You have presentational logic in here, with the error message that you want to display to the end user. I think any errors for the end user should probably go in the UI layer just so they're not spread out across your application.

If it's an error for a developer to read, I would rather you create a new  RuntimeException, maybe something like `class InvalidTelephoneNumber extends RuntimeException`, and you could bake into the custom exception class an error message that is more catered for a developer.

2. You're calling the validate from the constructor, which doesn't give the user much of a chance to deal with the error. They'd have to wrap the `new TelephoneNumber("foo")` in a try/catch. I tend to never put any logic in my constructor, I like constructors to only build the object I want to use. I just feel I can add logic later on easier.